core/src/main/java/org/apache/calcite/sql/fun/SqlTimestampAddFunction.java

```diff
@@ -129,8 +130,15 @@ static RelDataType deduceType(RelDataTypeFactory typeFactory,
     //    with startUnit = EPOCH and timeFrameName = 'MINUTE15'.
     //
     // If the latter, check that timeFrameName is valid.
-    validator.validateTimeFrame(
-        (SqlIntervalQualifier) call.getOperandList().get(0));
+    SqlIntervalQualifier op0 = call.operand(0);
+    validator.validateTimeFrame(op0);
+    if (op0.timeFrameName == null
```

Comment:
Is it possible to provide a method in `SqlValidatorImpl`  to do this check? But it looks good at the moment, maybe I'm too picky.